lib/teiserver/battle/balance/respect_avoids.ex

```diff
@@ -0,0 +1,529 @@
+        }
+  end
+
+  # This balance algorithm will use an adjusted rating for newish players
+  # This will not be displayed in chobby ui or player list; it's only used for balance
+  # It will be used when calculating team deviation
+  defp adjusted_rating(rating, uncertainty, rank) do
+    if(is_newish_player?(rank, uncertainty)) do
+      # For newish players we assume they are the worst in the lobby e.g. 0 match rating and
+      # then they converge to their true rating over time
+      # Once their uncertainty is low enough, we fully trust their rating
+      {_skill, starting_uncertainty} = Openskill.rating()
+
+      uncertainty_cutoff = @high_uncertainty
+
+      min(
+        1,
+        (starting_uncertainty - uncertainty) /
+          (starting_uncertainty - uncertainty_cutoff)
+      ) * rating
+    else
+      rating
+    end
+  end
+
+  @spec get_avoids([any()], number(), boolean()) :: [String.t()]
+  def get_avoids(player_ids, lobby_max_avoids, debug_mode? \\ false) do
+    # The max number of avoids per player to pull from db
+    player_limit = 2
+
+    cond do
+      debug_mode? ->
+        RelationshipLib.get_lobby_avoids(player_ids, lobby_max_avoids, player_limit)
+
+      true ->
+        avoid_min_hours = get_avoid_delay()
+
+        RelationshipLib.get_lobby_avoids(
+          player_ids,
+          lobby_max_avoids,
+          player_limit,
+          avoid_min_hours
+        )
+    end
+  end
+
+  @spec get_avoid_delay() :: number()
+  defp get_avoid_delay() do
+    Config.get_site_config_cache("lobby.Avoid min hours required")
+  end
+
+  ## Gets experienced players
+  def get_experienced_players(players, noobs) do
+    Enum.filter(players, fn player ->
+      !Enum.any?(noobs, fn noob ->
+        noob.name == player.name
+      end)
+    end)
+  end
+
+  ## Players that are in parties or avoids (in either direction) are at the front, then sort by rating
+  @spec sort_experienced_players([RA.player()], [[number()]]) :: [RA.player()]
+  def sort_experienced_players(experienced_players, avoids) do
+    flat_avoids = avoids |> List.flatten()
+
+    experienced_players
+    |> Enum.sort_by(
+      fn player ->
+        is_in_avoid_list? =
+          Enum.any?(flat_avoids, fn id ->
+            id == player.id
+          end)
+
+        avoid_relevance =
+          case is_in_avoid_list? do
+            true -> 100
+            false -> 0
+          end
+
+        party_relevance =
+          case player.in_party? do
+            true -> 1000
+            false -> 0
+          end
+
+        player.rating + avoid_relevance + party_relevance
+      end,
+      :desc
+    )
+  end
+
+  # Noobs have high uncertainty and chev 1,2,3
+  @spec get_noobs([RA.player()]) :: any()
+  def get_noobs(players) do
+    Enum.filter(players, fn player ->
+      is_newish_player?(player.rank, player.uncertainty)
+    end)
+  end
+
+  def is_newish_player?(rank, uncertainty) do
+    uncertainty >= @high_uncertainty && rank <= 2
```

Comment:
I think it's better to avoid using rank where possible, rank can be calculated by other methods than playtime in which case this stops working as intended.

Reply:
I won't change it for this PR as there are players who have high uncertainty that are 4 chevs who do not play like new players. New players do dumb stuff like make multiple labs and go t2 without resources. 4 chevs don't do that. 4 chevs can have high uncertainty from playing PvE, unranked, or different game modes like 1v1.

However, in the future I can look to improving this by giving playtime as a parameter to the balancer and could use that instead of rank.